scripts/link-packages.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+
+const { join } = require('path');
+const { readdirSync, statSync } = require('fs');
+const { spawnSync } = require('child_process');
+
+const args = process.argv.slice(2);
+
+const silent = args.find(x => x === '--silent');
+
+const currentFolder = process.cwd();
+
+const distFolder = join(currentFolder, '/dist');
+
+/**
+ * Checks name of the package to consider if the package is part of larger package/library
+ *
+ * @description
+ * The function is based on `npm` name convensions.
+ *
+ * @link https://docs.npmjs.com/cli/v6/using-npm/scope
+ * @link https://docs.npmjs.com/cli/v6/configuring-npm/package-json#name
+ *
+ * @param {*} packageJson The name of the package, extracted from package.json
+ *
+ * @returns {boolean} Package is root or not
+ */
+function isRootPackage(packageJson) {
+  if (!packageJson) {
+    throw new Error('Missing package.json');
+  }
+
+  if (!packageJson.name) {
+    throw new ReferenceError(`package.json doesn't have a 'name'`);
+  }
+
+  return packageJson.name.split('/').length <= 2;
+}
+
+/**
+ * Runs 'yarn link' inside package directory
+ *
+ * @param { name, file, dir } pkg Package that will be linked
+ * @returns {undefined}
+ */
+function linkPackage(pkg) {
+  if (!pkg) {
+    throw new Error('There is no package provided to link!');
+  }
+
+  if (!pkg.dir) {
+    throw new ReferenceError('Package directory location is missing!');
+  }
+
+  if (!pkg.name) {
+    throw new ReferenceError('Package name is missing!');
+  }
+
+  const spawn = spawnSync('yarn', ['link'], {
+    cwd: pkg.dir,
+  });
+
+  if (spawn.stdout.exit === 1) {
+    console.log(`${pkg.name} failed to link!`);
```

Comment:
I think that we have to throw an error and fail how task. 

Why?:
 * If one or more links could not be created we are not sure that the next build/compile task will pass at all (it may .. but it may not)
 * if there is no error `yarn/npm` won't throw an error and will suppress it inside the log.

I think this task is critical for the build so if it's not working for any reason there is no real reason for starting anything else.